### DISCUSSION TITLE
Web Components - Component nesting

### DISCUSSION THREAD

**Discussion by alizedebray** [swisspost/design-system#1508] [General]
How to validate that a parent component have the expected children components and vice versa
---
Current state: 
* the [post-tabs component](https://github.com/swisspost/design-system/pull/1181/files#diff-d496b51f79f260dff32758e2699de8958b13cb8f1f8a656fc3c414ed6b2a5e6e) performs multiple checks on whether it has the correct post-tab-header and post-tab-panel children

**Comment by gfellerph** [comment]
From my point of view, we don't have to do crazy checks to enforce a rigid system. I fear that setting up and maintaining these checks may outweigh their usefulness. What I'd like to do instead is patch stuff that can be easily forgotten:
- Add `slot="header"` to `post-tab-header` components
- Add `aria-describedby` attribute to light-dom triggers for the `post-tooltip` component

What I would not like to patch, because it's easy to spot the issue and to refer to the docs for remediation:
- People putting `<div>`s inside the `<post-tabs>` slot even if they don't belong there
